includes/orders/functions/refunds.php

```diff
@@ -342,6 +342,8 @@ function edd_refund_order( $order_id, $order_items = 'all', $adjustments = 'all'
 		if ( ! empty( $order_item['parent'] ) && ! empty( $order_item['original_item_status'] ) ) {
 			edd_update_order_item( $order_item['parent'], array( 'status' => $order_item['original_item_status'] ) );
 		}
+		edd_decrease_earnings( $order_item['product_id'], $order_item['amount'] );
+		edd_decrease_purchase_count( $order_item['product_id'], $order_item['quantity'] );
```

Comment:
I think we want to remove these. The download earnings specifically says "Gross Revenue", so it's not net.

Some discussion here: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/8659

Reply:
Although actually I see there is logic to decrease earnings in `edd_undo_purchase()`. So maybe the better move is to leave this addition in and open an entirely new issue to address this.

Reply:
If I take these out, though, and add the payment status hook for refunds back into that class, the download earnings numbers and counts start passing again.

However, the overall store earnings start to fail. In `test_refund_payment`, with those changes, I log this kind of information:
- `$store_earnings`: $360
- `$this->payment->total`: 120
- `edd_get_total_earnings()`: 120

The two failures specifically are:

> There were 2 failures:
> 
> 1) EDD\Orders\EDD_Payment_Tests::test_refund_payment
> Failed asserting that 120.0 matches expected 240.0.
> 
> /tmp/RWNMriU0H05xvvWnjeXJYHmrTcPWdM3b/tests/orders/tests-edd-payment.php:501
> 
> 2) EDD\Orders\EDD_Payment_Tests::test_refund_affecting_stats
> Failed asserting that 157.0 matches expected 277.0.

So....removing these lines and adding the extra hook in the payment class is some kind of progress.